core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -205,6 +202,21 @@ public void run() {
           }
         }
       }
+
+    private void rescheduleWithCause(Listener savedListener, Exception cause) {
```

Comment:
Note that this will cause some git conflicts with #4105. (It looks like it'll be unnecessary after that other PR.)

Reply:
Rebased and removed unnecessary rescheduling